songpal/method.py

```diff
@@ -152,18 +152,78 @@ async def __call__(self, *args, **kwargs):
 
     @property
     def inputs(self) -> Dict[str, type]:
-        """Input parameters for this method."""
-        return self.signature.input
+        """Input parameters for this method version."""
+        return (
+            self.signatures[self._version].input
+            if self.signatures[self._version].input is not None
+            else ""
+        )
+
+    @property
+    def latest_supported_version(self) -> str:
+        """Latest version supported by this method."""
+        return (
+            sorted(self._supported_versions, reverse=True)[0]
+            if len(self._supported_versions) > 0
+            else None
+        )
 
     @property
     def outputs(self) -> Dict[str, type]:
-        """Output parameters for this method."""
-        return self.signature.output
+        """Output parameters for this method version."""
+        return (
+            self.signatures[self._version].output
+            if self.signatures[self._version].output is not None
+            else ""
```

Comment:
I think `return self.signatures[self._version].output or ""` should work here.

Reply:
I think when I originally did this I had removed `inputs` and `outputs`. I don't think this is really needed.